### DISCUSSION TITLE
Naming of SysTick Registers

### DISCUSSION THREAD

**Discussion by Masmiseim36** [ARM-software/CMSIS_5#1628] [General]
Hello Community,

why are the SystemTick registers defined like this
typedef struct
{
  __IOM uint32_t CTRL;                   /*!< Offset: 0x000 (R/W)  SysTick Control and Status Register */
  __IOM uint32_t LOAD;                   /*!< Offset: 0x004 (R/W)  SysTick Reload Value Register */
  __IOM uint32_t VAL;                    /*!< Offset: 0x008 (R/W)  SysTick Current Value Register */
  __IM  uint32_t CALIB;                  /*!< Offset: 0x00C (R/ )  SysTick Calibration Register */
} SysTick_Type;
In the CMSIS library while they are defined as CSR, RVR, CVR and CALIB in the ARM Architecture Reference Manual?
https://developer.arm.com/documentation/ddi0403/d/System-Level-Architecture/System-Address-Map/The-system-timer--SysTick/System-timer-register-support-in-the-SCS?lang=en

Other registers seem to be named according to the definition from the Reference Manual.
It is a bit confusing if they are named differently in the register view of the debugger and in the code.
Is there a particular reason for the difference?

Regards

**Comment by JonatanAntoni** [comment]
Hi @Masmiseim36,

This is for historic reasons. As a rule of thumb we want to use register names similar to the manual. In some areas we want to be consistent across different versions and implementations of the architecture and face renames. Hence, it can always happen, that we have slightly different names in CMSIS compared with (different versions) if the manuals. 

Cheers,
Jonatan

**Comment by Masmiseim36** [comment]
Hello Jonatan,

Thanks for clarification.
Since you are working on the next major version of CMSIS, would this be a good time to correct this mismatch?
Or you could add the version of to the references manual in parallel to the legacy version. Something like this:

```
/**
  \brief  Structure type to access the System Timer (SysTick).
 */
typedef struct
{
  union
  {
    CMSIS_DEPRECATED __IOM uint32_t CTRL; /*!< Offset: 0x000 (R/W)  SysTick Control and Status Register */
	__IOM uint32_t CSR;
  };
  union
  {
    CMSIS_DEPRECATED __IOM uint32_t LOAD; /*!< Offset: 0x004 (R/W)  SysTick Reload Value Register */
	__IOM uint32_t RVR;
  };
  union
  {
    CMSIS_DEPRECATED __IOM uint32_t VAL; /*!< Offset: 0x008 (R/W)  SysTick Current Value Register */
    __IOM uint32_t CVR;
  };
  __IM  uint32_t CALIB;                  /*!< Offset: 0x00C (R/ )  SysTick Calibration Register */
} SysTick_Type;
```


Regards
